code/modules/surgery/organs/liver.dm

```diff
@@ -21,6 +21,46 @@
 	var/toxLethality = LIVER_DEFAULT_TOX_LETHALITY//affects how much damage toxins do to the liver
 	var/filterToxins = TRUE //whether to filter toxins
 
+/obj/item/organ/liver/Initialize()
+	. = ..()
+	// If the liver handles foods like a clown, it honks like a bike horn
+	// Don't think about it too much.
+	RegisterSignal(src, SIGNAL_ADDTRAIT(TRAIT_COMEDY_METABOLISM), .proc/on_comedy)
+	RegisterSignal(src, SIGNAL_REMOVETRAIT(TRAIT_COMEDY_METABOLISM), .proc/on_comedy_remove)
+
+/obj/item/organ/liver/proc/on_comedy()
+	// Are clown "bike" horns made from the livers of ex-clowns?
+	// Would that make the clown more or less likely to honk it
+	AddComponent(/datum/component/squeak, list('sound/items/bikehorn.ogg'=1), 50, falloff_exponent = 20)
+
+/obj/item/organ/liver/proc/on_comedy_remove()
+	qdel(GetComponent(/datum/component/squeak))
```

Comment:
Add an initialization parameter for the squeak component to register to signal removal of the comedy trait metabolism, and delete itself.
Avoid `GetComponent` for manipulating components.

Reply:
I'm not sure if I undestand what you want?

Add an initialization parameter to the squeak component, passing in a trait that it needs to watch the parent for, and then self delete when that trait is removed? That doesn't really seem like an improvement in code quality, which is why I feel like I've misunderstood it.

Reply:
The squeak component registers to signals based on the parent type.
You'll want to register to the comedy removal here, for liver types, and self-delete.
Components should handle their own deletion.